### ISSUE TITLE
twister: regression : Failures are counted as errors

### ISSUE THREAD

**Issue by tpennors** [zephyrproject-rtos/zephyr#47702]
**Describe the bug**
Failures are counted as errors. Must be a regression because it was working well before

**To Reproduce**
1. Create a build error in `zephyr/samples/hello_world/main.c`
2. run ./scripts/twister -T samples/hello_world/ -p b_u585i_iot02a
3. See error:
![twister_error](https://user-images.githubusercontent.com/15906660/178548347-3aba1ed4-aa71-44e0-87b5-0bada2227a0c.PNG)

**Impact**
The big problem is that as failure and error are inverted, twister return code is not 1 when failure occurs so CI does not pass in failure. As it can be seen in twister script:     
```
if results.failed or (tplan.warnings and options.warnings_as_errors):
        sys.exit(1)
```

**Environment (please complete the following information):**
 - OS: OS: ubuntu 20.04
 - Toolchain zephyr-sdk 0.13.1
 - Commit SHA or Version : v3.1.0 TAG

**Comment by nashif** [comment]
backport fix available, this is 3.1 specific.

**Comment by nashif** [comment]
https://github.com/zephyrproject-rtos/zephyr/pull/47703 merged
